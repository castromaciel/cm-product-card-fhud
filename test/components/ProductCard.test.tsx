import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { product1 } from '../data/data';

const { act } = renderer;

describe('ProductCard', () => {

  test('debe mostrar el componente correctamente', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          () => (
            <h1>ProductCard</h1>
          )
        }
      </ProductCard>
    )

    expect( wrapper.toJSON() ).toMatchSnapshot()

  })

  test('debe incrementar el contador', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          ({count, increaseBy}) => (
            <>
              <h1>ProductCard</h1>
              <span>{count} <p>hola</p></span>
              <button onClick={ () => increaseBy(1)}></button>
            </>
          )
        }
      </ProductCard>
    );
    let tree = wrapper.toJSON();
    console.log(tree);

    expect( tree ).toMatchSnapshot();

    act( () => {
      (tree as any).children[2].props.onClick();
    })

    tree = wrapper.toJSON();
    
    expect((tree as any).children[1].children[0]).toBe('1');
    
    expect( tree ).toMatchSnapshot()
  }
)

})