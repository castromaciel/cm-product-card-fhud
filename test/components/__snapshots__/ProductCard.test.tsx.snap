// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard debe incrementar el contador 1`] = `
<article
  className=" productCard undefined"
>
  <h1>
    ProductCard
  </h1>
  <span>
    0
     
    <p>
      hola
    </p>
  </span>
  <button
    onClick={[Function]}
  />
</article>
`;

exports[`ProductCard debe incrementar el contador 2`] = `
<article
  className=" productCard undefined"
>
  <h1>
    ProductCard
  </h1>
  <span>
    1
     
    <p>
      hola
    </p>
  </span>
  <button
    onClick={[Function]}
  />
</article>
`;

exports[`ProductCard debe mostrar el componente correctamente 1`] = `
<article
  className=" productCard undefined"
>
  <h1>
    ProductCard
  </h1>
</article>
`;
